File: seminar/src/main/java/com/server/dosopt/seminar/controller/ServiceMemberController.java
package com.server.dosopt.seminar.controller;

import com.server.dosopt.seminar.dto.request.servicemember.ServiceMemberRequest;
import com.server.dosopt.seminar.dto.response.servicemember.ServiceMemberResponse;
import com.server.dosopt.seminar.service.ServiceMemberService;
import java.net.URI;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/users/")
public class ServiceMemberController {

    private final ServiceMemberService serviceMemberService;

    @PostMapping("sign-up")
    public ResponseEntity<Void> signUp(@RequestBody ServiceMemberRequest request) {
        URI location = URI.create(serviceMemberService.create(request));
        return ResponseEntity.created(location).build();
    }

    @PostMapping("sign-in")
    public ResponseEntity<ServiceMemberResponse> signIn(@RequestBody ServiceMemberRequest request) {
        return ResponseEntity.ok().body(serviceMemberService.signIn(request));
    }
}

File: kotlin-seminar/src/main/kotlin/sopt/org/kotlinseminar/controller/ServiceMemberController.kt
package sopt.org.kotlinseminar.controller

import lombok.RequiredArgsConstructor
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import sopt.org.kotlinseminar.dto.request.servicemember.ServiceMemberRequest
import sopt.org.kotlinseminar.service.ServiceMemberService
import java.net.URI


@RestController
@RequiredArgsConstructor
@RequestMapping("/api/users/")
class ServiceMemberController(
        private val serviceMemberService: ServiceMemberService
) {
    @PostMapping("sign-up")
    fun signUp(@RequestBody request: ServiceMemberRequest): ResponseEntity<Void> {
        val location: URI = URI.create(serviceMemberService.create(request))
        return ResponseEntity.created(location).build()
    }

    @PostMapping("sign-in")
    fun signIn(@RequestBody request: ServiceMemberRequest): ResponseEntity<Void> {
        serviceMemberService.signIn(request)
        return ResponseEntity.noContent().build()
    }
}